session_3/blogspot/app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,86 @@
+class ArticlesController < ApplicationController
+  load_and_authorize_resource
+
+  before_action :set_article, only: [:show, :edit, :update, :destroy]
+
+  def article_params
+    params.require(:article).permit(:title, :topic, :tags, :content, :public)
+  end
+
+  # GET /articles or /articles.json
+  def index
+    @articles = Article.all
+    if !current_user
+      session[:private_access] = 5
+    end
+  end
+
+  # GET /articles/1 or /articles/1.json
+  def show
+    if current_user
+     if can? :update, @article, user_id: current_user.id || current_user.admin
```

Comment:
Why is `can? :update` relevant to show action?

Reply:
We have the `Edit` and `Destroy` links beside the Show link, so I want to hide them so we avoid `Access Violation`